ZuluContent/Items/Clothing/Hats/BaseHat.cs

```diff
@@ -1,62 +1,57 @@
 using Server.Network;
 using System.Collections.Generic;
+using ZuluContent.Zulu.Engines.Magic.Enchantments;
+using ZuluContent.Zulu.Engines.Magic.Enums;
 
 namespace Server.Items
 {
-	public abstract class BaseHat : BaseClothing, IShipwreckedItem
-	{
-		private bool m_IsShipwreckedItem;
-
-		[CommandProperty( AccessLevel.GameMaster )]
-		public bool IsShipwreckedItem
-		{
-			get { return m_IsShipwreckedItem; }
-			set { m_IsShipwreckedItem = value; }
-		}
-
-		public BaseHat( int itemID ) : this( itemID, 0 )
-		{
-		}
-
-		public BaseHat( int itemID, int hue ) : base( itemID, Layer.Helm, hue )
-		{
-		}
-
-		public BaseHat( Serial serial ) : base( serial )
-		{
-		}
-
-		public override void Serialize( IGenericWriter writer )
-		{
-			base.Serialize( writer );
-
-			writer.Write( (int) 1 ); // version
-
-			writer.Write( m_IsShipwreckedItem );
-		}
-
-		public override void Deserialize( IGenericReader reader )
-		{
-			base.Deserialize( reader );
-
-			int version = reader.ReadInt();
-
-			switch ( version )
-			{
-				case 1:
-				{
-					m_IsShipwreckedItem = reader.ReadBool();
-					break;
-				}
-			}
-		}
-
-		public override void AddEquipInfoAttributes( Mobile from, List<EquipInfoAttribute> attrs )
-		{
-			base.AddEquipInfoAttributes( from, attrs );
-
-			if( m_IsShipwreckedItem )
-				attrs.Add( new EquipInfoAttribute( 1041645 ) );	// recovered from a shipwreck
-		}
-	}
-}
+    public abstract class BaseHat : BaseClothing
+    {
+        [CommandProperty(AccessLevel.GameMaster)]
+        public HatFortification Fortified
+        {
+            get => Enchantments.Get((ItemFortification e) => (HatFortification) e.Value);
+            set => Enchantments.Set((ItemFortification e) => e.Value = (int) value);
+        }
+
+        public override double ArmorScalar => Fortified == HatFortification.Fortified ? 0.15 : 0.5;
+
+        public BaseHat(int itemID, int hue) : base(itemID, Layer.Helm, hue)
+        {
+        }
+
+        public BaseHat(Serial serial) : base(serial)
+        {
+        }
+
+        public void Fortify(BaseArmor helm)
+        {
+            MaxHitPoints = helm.MaxHitPoints;
+            HitPoints = helm.HitPoints;
+            BaseArmorRating = helm.BaseArmorRating;
+
+            foreach (var (_, value) in helm.Enchantments.Values)
+            {
+                Enchantments.SetFromFortified(value);
```

Comment:
You can use just `Enchantment.Set(value)` here since the helm is being deleted, otherwise you would need to clone the object.